webvh/webvh/did/operations_manager.py

```diff
@@ -0,0 +1,496 @@
+        self.profile = profile
+
+    def _all_are_not_none(*args):
+        return all(v is not None for v in args)
+
+    async def fetch_jsonl(self, url):
+        """Fetch a JSONL file from the given URL."""
+        async with ClientSession() as session:
+            async with session.get(url, ssl=False) as response:
+                # Check if the response is OK
+                response.raise_for_status()
+
+                # Read the response line by line
+                async for line in response.content:
+                    # Decode each line and parse as JSON
+                    decoded_line = line.decode("utf-8").strip()
+                    if decoded_line:  # Ignore empty lines
+                        yield json.loads(decoded_line)
+
+    async def _request_identifier(self, server_url, namespace, identifier) -> tuple:
+        """Contact the trust did web server to request an identifier."""
+        async with ClientSession() as session:
+            try:
+                response = await session.get(
+                    server_url,
+                    params={
+                        "namespace": namespace,
+                        "identifier": identifier,
+                    },
+                    ssl=False,
+                )
+            except ClientConnectionError as err:
+                raise DidCreationError(f"Failed to connect to Webvh server: {err}")
+
+            response_json = await response.json()
+            if response.status == http.HTTPStatus.BAD_REQUEST:
+                raise DidCreationError(response_json.get("detail"))
+
+            document = response_json.get("didDocument", {})
+            did = document.get("id")
+
+            proof_options = response_json.get("proofOptions", {})
+            challenge = proof_options.get("challenge")
+            domain = proof_options.get("domain")
+            expiration = proof_options.get("expires")
+
+            if self._all_are_not_none(did, challenge, domain, expiration):
+                return did, challenge, domain, expiration
+            else:
+                raise DidCreationError(
+                    "Invalid response from Webvh server requesting identifier"
+                )
+
+    async def _get_or_create_author_key(self, did):
+        async with self.profile.session() as session:
+            try:
+                key_info_author = await MultikeyManager(session).create(
+                    alg="ed25519",
+                    kid=f"{did}#author",
+                )
+            except MultikeyManagerError:
+                key_info_author = await MultikeyManager(session).from_kid(f"{did}#author")
+
+        return key_info_author
+
+    async def _create_author_signed_registration_document(
+        self, did, author_key_info, expiration, domain, challenge
+    ):
+        async with self.profile.session() as session:
+            return await DataIntegrityManager(session).add_proof(
+                DIDDocument(
```

Comment:
I'm ok with this, but this is making the decision to add the authorized key as a verification method. This may or may not be desirable for security reasons. I'm ok with leaving it as is and including a comment to revisit this and discuss with the acapy community.